### PR TITLE
kind: Allow to install custom kubevirt versions

### PR DESCRIPTION
**Pull request by oshoval**
#### What this PR does and why is it needed
Allow to install custom Kubevirt releases.
Also allows to freeze specific version upon needs, for example if newest breaks stuff,
or if a feature is required before a stable release exists.

Default stays deploy last stable as it was.
Since nightly doesn't release virtctl binary, keep using the latest stable released virtctl.

Use by exporting optional env var `KUBEVIRT_VERSION` with the requested value.

Possible values of `KUBEVIRT_VERSION`:
```
stable - install newest stable (default)
vX.Y.Z - install specific stable (i.e v1.3.1)
nightly - install newest nightly
nightly tag - install specific nightly (i.e 20240910)
```

#### Which issue(s) this PR fixes

Fixes #

#### Special notes for reviewers

`quay.io/kubevirt/network-passt-binding` does have nightly,
in order to determine the tag, some more logic need to be added,
so for now only support stable

Note that the daemonset that deploys the passt binary use only stable releases,
but this is not due to this PR.

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

**Comment by oshoval** [comment]
/cc @maiqueb @qinqon 
please take a look once you can, thanks

File: contrib/kind-common [hunk 1]
```diff
@@ -320,7 +320,23 @@ is_nested_virt_enabled() {
 }
 
 install_kubevirt() {
-    local kubevirt_version="$(curl -L https://storage.googleapis.com/kubevirt-prow/release/kubevirt/kubevirt/stable.txt)"
+    # possible values:
+    # stable - install newest stable (default)
+    # vX.Y.Z - install specific stable (i.e v1.3.1)
+    # nightly - install newest nightly
+    # nightly tag - install specific nightly (i.e 20240910)
```
Comment (RamLavi) [inline contrib/kind-common:327]: Can you add a link in the comment to where one could look for specific nightlies?

File: contrib/kind-common [hunk 2]
```diff
@@ -320,7 +320,23 @@ is_nested_virt_enabled() {
...
+
+    if [[ "$KUBEVIRT_VERSION" == "stable" ]] || [[ "$KUBEVIRT_VERSION" == v* ]]; then
+        local kubevirt_version=$( [[ "$KUBEVIRT_VERSION" == "stable" ]] && curl -sL https://storage.googleapis.com/kubevirt-prow/release/kubevirt/kubevirt/stable.txt || echo "$KUBEVIRT_VERSION" )
+        echo "Deploy stable Kubevirt release $kubevirt_version"
+        local kubevirt_release_url="https://github.com/kubevirt/kubevirt/releases/download/${kubevirt_version}"
+    else
+        local kubevirt_version=$( [[ "$KUBEVIRT_VERSION" == "nightly" ]] && curl -sL https://storage.googleapis.com/kubevirt-prow/devel/nightly/release/kubevirt/kubevirt/latest || echo "$KUBEVIRT_VERSION" )
+        echo "Deploy nightly Kubevirt build $kubevirt_version"
+        local kubevirt_release_url="https://storage.googleapis.com/kubevirt-prow/devel/nightly/release/kubevirt/kubevirt/${kubevirt_version}"
+    fi
```
Comment (qinqon) [inline contrib/kind-common:338]: Let's just put in if statements the three use cases, even if two of the are going to have the same release URL base, it makes code cleaner and reduce inlining

```suggestion
    if [[ "$KUBEVIRT_VERSION" == "stable" ]]; then
        local kubevirt_version=$( curl -sL https://storage.googleapis.com/kubevirt-prow/release/kubevirt/kubevirt/stable.txt )
        local kubevirt_release_url="https://github.com/kubevirt/kubevirt/releases/download/${kubevirt_version}"
    elif  [[ "$KUBEVIRT_VERSION" == "v*" ]]; then
          local kubevirt_version=$KUBEVIRT_VERSION      
          local kubevirt_release_url="https://github.com/kubevirt/kubevirt/releases/download/${kubevirt_version}"
    elif [[ $KUBEVIRT_VERSION" == "nightly" ]]; then  
           local kubevirt_version=$( curl -sL https://storage.googleapis.com/kubevirt-prow/devel/nightly/release/kubevirt/kubevirt/latest )
           local kubevirt_release_url="https://storage.googleapis.com/kubevirt-prow/devel/nightly/release/kubevirt/kubevirt/${kubevirt_version}"
    else
        // error
    fi
```

**Review by qinqon — CHANGES_REQUESTED** [review]
Let's remove inlining and expand them.

File: contrib/kind-common (hunk 1 above)
Comment (oshoval) [reply contrib/kind-common:327]: only thing i know is taking the date part from https://quay.io/repository/kubevirt/virt-handler?tab=tags&tag=latest

i prefer to not comment it in the file itself, it aint clean

File: contrib/kind-common (hunk 2 above)
Comment (oshoval) [reply contrib/kind-common:338]: it miss one case of specific date tag, but i will try to cover it with a regex of 6 digits

File: contrib/kind-common (hunk 1 above)
Comment (RamLavi) [reply contrib/kind-common:327]: yeah, if they don't have a proper API for it, then not needed..
I was thinking they may have something like our cnv-explorer or something..
so all good

File: contrib/kind-common (hunk 2 above)
Comment (qinqon) [reply contrib/kind-common:338]: Important part is to remove inlining and create three if statements to make things clear for future selves with super dummy easy bash code.

**Comment by oshoval** [comment]
addressed comments, also created a function for it

**Comment by qinqon** [comment]
LGTM

**Comment by oshoval** [comment]
> LGTM

Thanks Quique, I think an approval via the review console is required please ? so it will be visible on the top right reviewers status list

**Review by RamLavi — APPROVED** [review]
/lgtm

**Comment by oshoval** [comment]
rebased

**Comment by oshoval** [comment]
added on PR desc info about `quay.io/kubevirt/network-passt-binding` that was added lately

**Comment by oshoval** [comment]
rebased

@qinqon can you please take another look  ?

**Review by qinqon — APPROVED** [review]
LGTM

**Comment by maiqueb** [comment]
/cc @jcaamano 

could you take a quick look at this ? We are hoping to get this merged to improve the dev experience.
Thanks in advance.

**Review by jcaamano — DISMISSED** [review]

**Comment by oshoval** [comment]
closing and reopening so it will run again actions
thx

**Comment by jcaamano** [comment]
@oshoval could you push again? I generated a merge commit unintentionally.

**Comment by oshoval** [comment]
done, also cleaned white space
thx

**Review by jcaamano — APPROVED** [review]

**Pull request merged**